src/audio/pipeline.c

```diff
@@ -616,6 +616,46 @@ static void pipeline_comp_trigger_sched_comp(struct pipeline *p,
 	list_item_append(&p->list, &ctx->pipelines);
 }
 
+/*
+ * Check whether pipeline is incapable of acquiring data for capture.
+ *
+ * If capture START/RELEASE trigger originated on dailess pipeline and reached
+ * inactive pipeline as it's source, then we indicate that it's blocked.
+ *
+ * @param rsrc - component from remote pipeline serving as source to relevant
+ *		 pipeline
+ * @param ctx - trigger walk context
+ * @param dir - trigger direction
+ */
+static inline bool pipeline_has_no_data(struct comp_dev *rsrc,
+					struct pipeline_walk_context *ctx,
+					int dir)
+{
+	struct pipeline_data *ppl_data = ctx->comp_data;
+	struct comp_dev *pipe_source = ppl_data->start->pipeline->source_comp;
+
+	/* only applies to capture pipelines */
+	if (dir != SOF_IPC_STREAM_CAPTURE)
+		return false;
+
+	/* only applicable on trigger start/release */
+	if (ppl_data->cmd != COMP_TRIGGER_START &&
```

Comment:
Not sure if the trigger should be tested inside the function which checks whether capture source is ready.  This function would be more universal if this specific condition is tested by the caller.